delphi_srr2_radar_driver_wrapper/src/delphi_srr2_radar_driver_wrapper.cpp

```diff
@@ -15,24 +15,61 @@
  */
 
 #include "delphi_srr2_radar_driver_wrapper.h"
+#include <radar_msgs/RadarTrackArray.h>
+#include <radar_msgs/RadarStatus.h>
 
 DelphiSrr2RadarDriverWrapper::DelphiSrr2RadarDriverWrapper(int argc, char **argv, const std::string &name) : DriverWrapper (argc, argv, name) {}
 
 DelphiSrr2RadarDriverWrapper::~DelphiSrr2RadarDriverWrapper() {}
 
 void DelphiSrr2RadarDriverWrapper::initialize() {
-    srr_status5_sub_ = nh_->subscribe("srr_status5", 10, &DelphiSrr2RadarDriverWrapper::srr_status5_cb, this);
+
+    // Set driver type
     status_.sensor = true;
+
+    // Initilize all subscribers for SRR driver
+    detection_sub_ = nh_->subscribe("detections", 10, &DelphiSrr2RadarDriverWrapper::detection_cb, this);
+    status1_sub_ = nh_->subscribe("srr_status1", 10, &DelphiSrr2RadarDriverWrapper::status1_cb, this);
+    status2_sub_ = nh_->subscribe("srr_status2", 10, &DelphiSrr2RadarDriverWrapper::status2_cb, this);
+    status5_sub_ = nh_->subscribe("srr_status5", 10, &DelphiSrr2RadarDriverWrapper::status5_cb, this);
+
+    // Initilize all publishers for CARMA
+    object_track_pub_ = nh_->advertise<radar_msgs::RadarTrackArray>("radar/srr/tracks_raw", 1);
+    radar_status_pub_ = nh_->advertise<radar_msgs::RadarStatus>("radar/srr/status", 1);
+
+    private_nh_->param<std::string>("sensor_frame", sensor_frame_, "srr2");
+    private_nh_->param<double>("bounding_box_size", bounding_box_size_, 5);
+
+    last_update_time = ros::Time::now();
 }
 
-void DelphiSrr2RadarDriverWrapper::pre_spin() {}
+void DelphiSrr2RadarDriverWrapper::pre_spin()
+{
+    checkRadarTimeout();
+    publish_radar_status();
+    publish_object_track();
```

Comment:
> You want to publish in post spin. Spin will call all your subscribers to update the current message variables.
> Also you'll want a rate of 50Hz since that is probably the radar publication rate.

I totally forget about the spin() will trigger the subscribers callbacks... Thanks.